### DISCUSSION TITLE
Purged characters on new UI file names

### DISCUSSION THREAD

**Discussion by Gilbum** [aclap-dev/video-downloadhelper#1252] [Help and Q&A]
### Your question

I like very much the new UI and I have only one issue before have it as my default. The problem is about the too many characters purged in the suggested file names: brackets  round or square for example. I do not find any place where eventually customize this option. The old UI allows all characters (just purges asterisk or question mark as obvious).
Is there something I didn't find?
Thanks 
Gilberto

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: it
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"ec3b50a367d74f14"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "J:\\jap\\yy",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "J:\\jap\\yy"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "default_action": "download_as",
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 2,
    "successfull_dl": 92,
    "never_show_successfull_dl_message": true,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_6032039830845459"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "J:\\jap\\yy\\Mago Italy 2000 - Monica Roccaforte - EPORNER.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Mago Italy 2000 - Monica Roccaforte - EPORNER.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "J:\\jap\\yy"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.eporner.com/video-ZjIAtN3UQte/mago-italy-2000/"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1726162656987
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I've been noticing this for a while now.  But this affects only the default name offered by VDH.  This is a default.  That means you don't have to accept it.  Also, it is offered.  That means you don't have to accept it.  Remember, it's a default offer, not a forced value.  You should switch your default action away from `Download` & use `Download As` as your default action.  Then you will be able to fix your file names to be anything you want.

I may be guessing at another issue, somewhat related here.  But you might want to take a look at #838.  There's a lot of discussion in there on a topic you might not be interested in.  But VDH is set to a default that limits the length of the file name that it offers as the default.  There's that phrase again, default offer.  You don't have to accept it.  Read the first few posts in #838, the ones with a June 7 date.  You will encounter little things that GitHub hides with `Show x previous replies` or `Show x following replies`.  Be sure you click those to reveal the posts GitHub has decided to hide.  You will see some images showing you how to change the default file name length.

I realize this might not be the most convenient way to do things.  But until the issue gets fixed, this is what you will have to do.

**Comment by Gilbum** [comment]
Yes, I got it. Thanks for all things you said.
I'd like to leave as **enhancement request the option to modify the rejected characters in default file naming**. As you said in another discussion ([#1149](https://github.com/aclap-dev/video-downloadhelper/discussions/1149)), Windows (that I think has more restrictions than Mac and Linux) do not allow / \ | : * ? " < > in file names, _but why filter down all other special chars_?
Thanks again
